bolt-sidecar/src/state/execution.rs

```diff
@@ -269,89 +269,95 @@ impl<C: StateFetcher> ExecutionState<C> {
             return Err(ValidationError::BaseFeeTooLow(max_basefee));
         }
 
-        // From previous preconfirmations requests retrieve
-        // - the nonce difference from the account state.
-        // - the balance difference from the account state.
-        // - the highest slot number for which the user has requested a preconfirmation.
-        //
-        // If the templates do not exist, or this is the first request for this sender,
-        // its diffs will be zero.
-        let (nonce_diff, balance_diff, highest_slot) = self.block_templates.iter().fold(
-            (0, U256::ZERO, 0),
-            |(nonce_diff_acc, balance_diff_acc, highest_slot), (slot, block_template)| {
-                let (nonce_diff, balance_diff, slot) = block_template
-                    .get_diff(&sender)
-                    .map(|(nonce, balance)| (nonce, balance, *slot))
-                    .unwrap_or((0, U256::ZERO, 0));
-
-                (
-                    nonce_diff_acc + nonce_diff,
-                    balance_diff_acc.saturating_add(balance_diff),
-                    u64::max(highest_slot, slot),
-                )
-            },
-        );
-
-        if target_slot < highest_slot {
```

Comment:
@thedevbirb unsure why this error is only thrown when the target slot is smaller than the highest slot in the template. It should be thrown every time the target slot is lower than the current one, which is stored in `self.slot` right?

Reply:
In general yes we should also check that `target_slot >= self.slot`. I thought we were doing that already (I could be wrong)

The idea of that check is the following: assume I have nonce $N$, I sent a preconf for slot $N+2$ and then a preconf for slot $N+1$. In theory, the latter is valid but it invalidates the preconf for slot $N+2$. To avoid managing these edge cases, we simply require that preconfirmations in a increasing order of `target_slot`.
Does it make sense in your opinion? cc @merklefruit since we discussed about this together a while ago